rmw_fastrtps_shared_cpp/src/rmw_publisher.cpp

```diff
@@ -101,4 +101,61 @@ __rmw_publisher_count_matched_subscriptions(
 
   return RMW_RET_OK;
 }
+
+rmw_ret_t
+__rmw_publisher_get_actual_qos(
+  const rmw_publisher_t * publisher,
+  rmw_qos_profile_t * qos)
+{
+  RMW_CHECK_ARGUMENT_FOR_NULL(publisher, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_ARGUMENT_FOR_NULL(qos, RMW_RET_INVALID_ARGUMENT);
+
+  auto info = static_cast<CustomPublisherInfo *>(publisher->data);
+  if (info == nullptr) {
+    return RMW_RET_ERROR;
+  }
+  eprosima::fastrtps::Publisher * fastrtps_pub = info->publisher_;
+  if (fastrtps_pub == nullptr) {
+    return RMW_RET_ERROR;
+  }
+  const eprosima::fastrtps::PublisherAttributes attributes =
```

Comment:
@ivanpauno just curious, can we get this one as a `const ref` instead?

Reply:
Yes, I'll modify it.